src/dc_mimefactory.rs

```diff
@@ -230,42 +244,24 @@ pub unsafe fn dc_mimefactory_load_msg(
         }
     }
 
-    factory.loaded = DC_MF_MSG_LOADED;
+    factory.loaded = Loaded::Message;
     factory.timestamp = factory.msg.timestamp_sort;
-    factory.rfc724_mid = dc_strdup(factory.msg.rfc724_mid);
+    factory.rfc724_mid = {
+        // XXX: can factory.msg.rfc_724_mid be NULL at this point?
+        let s = dc_strdup(factory.msg.rfc724_mid);
```

Comment:
no need to call `dc_strdup` on this if you call `to_string` on it.

Reply:
Done.